backend-models/service-tests/sectionedservice-test.txt

```diff
@@ -0,0 +1,551 @@
+sectionedservice.php
+
+----------------------------------
+getGroupsTest() line 100:
+----------------------------------
+
+*********
+ Pre-req
+*********
+
+(checklogin) IS TRUE
+
+//// PRE-REQ Values /////
+
+send
+{
+$opt = "GRP"
+$courseid = "1"
+$coursevers = "45656"
+}
+
+/////////////////////////
+
+
+*****************
+ Compare results
+*****************
+
+If $debug is empty then successfully gathered all groups
+
+*****************
+ Service output
+*****************
+
+This service gather groups and if the groups change (unclear in code)
+this will affect:
+
+"groups",
+"grpmembershp",
+"grplst"
+
+----------------------------------
+deleteTest() line 181:
+----------------------------------
+
+*********
+ Pre-req
+*********
+
+(checklogin) IS TRUE
+AND $ha || $studentTeacher IS TRUE
+
+//// PRE-REQ Values /////
+
+send
+{
+$opt = "DEL"
+$sectid != null
+}
+
+/////////////////////////
+
+You also need to gather all listentries to compare if deleted.
+SQL Query: "SELECT FROM listentries WHERE lid=:lid" or "SELECT FROM listentries" save into beforeDeleted
+
+**************************
+ Get deleted listentries
+**************************
+
+SQL Query: "SELECT FROM listentries WHERE lid=:lid" or "SELECT FROM listentries" (depends on which was used earlier) save into afterDeleted
+
+*****************
+ Compare results
+*****************
+
+If beforeDeleted(count) > afterDeleted(count): passed
+If beforeDeleted(count) < afterDeleted(count): failed
+If $debug not empty: failed
+
+*****************
+ Service output
+*****************
+
+"codeexamples"
+
```

Comment:
When the test has been executed, the database is to be reverted to the state it was before.
i would recommend creating an insert that you later deleted trough this test. (or if the test fail by a query)

Reply:
this is true for ALL tests, therefore i will not write this in every test. you just need to make sure that this is true in every test.